lib/rouge/lexers/csharp.rb

```diff
@@ -43,7 +43,7 @@ class CSharp < RegexLexer
 
       state :whitespace do
         rule /\s+/m, Text
-        rule %r(//.*?\n), Comment::Single
+        rule %r(//.*?$), Comment::Single
```

Comment:
Use the end-of-line regex symbol (`$`) instead of requiring a newline character (`\n`).